climada/entity/exposures/litpop/litpop.py

```diff
@@ -0,0 +1,1108 @@
+                                        data_dir, resample_first
+                                        )
+                           for idc, country in enumerate(countries)]
+
+        else: # else, as default, country is initiated as a whole:
+            # loop over countries: litpop is initiated for each individual polygon
+            # within each country and combined at the end.
+            litpop_list = \
+                [self._set_one_country(country,
+                                       res_arcsec=res_arcsec,
+                                       exponents=exponents,
+                                       fin_mode=fin_mode,
+                                       total_value=total_values[idc],
+                                       reference_year=reference_year,
+                                       gpw_version=gpw_version,
+                                       data_dir=data_dir,
+                                       resample_first=resample_first)
+                 for idc, country in enumerate(countries)]
+        # make lists of countries with Exposure initaited and those ignored:
+        countries_in = \
+            [country for i, country in enumerate(countries) if litpop_list[i] is not None]
+        countries_out = \
+            [country for i, country in enumerate(countries) if litpop_list[i] is None]
+        if not countries_in:
+            raise ValueError('No valid country identified in %s, aborting.' % countries)
+        if countries_out:
+            LOGGER.warning('Some countries could not be identified and are ignored: '
+                           '%s. Litpop only initiated for: %s'
+                           % (countries_out, countries_in))
+
+        tag.description = ('LitPop Exposure for %s at %i as, year: %i, financial mode: %s, '
+                           'exp: [%i, %i], admin1_calc: %s'
+                           % (countries_in, res_arcsec, reference_year, fin_mode,
+                              exponents[0], exponents[1], str(admin1_calc)))
+
+        Exposures.__init__(
+            self,
+            data=Exposures.concat(litpop_list).gdf,
+            crs=litpop_list[0].crs,
+            ref_year=reference_year,
+            tag=tag,
+            value_unit=value_unit
+        )
+
+        try:
+            rows, cols, ras_trans = u_coord.pts_to_raster_meta(
+                (self.gdf.longitude.min(), self.gdf.latitude.min(),
+                 self.gdf.longitude.max(), self.gdf.latitude.max()),
+                u_coord.get_resolution(self.gdf.longitude, self.gdf.latitude))
+            self.meta = {
+                'width': cols,
+                'height': rows,
+                'crs': self.crs,
+                'transform': ras_trans,
+            }
+        except ValueError:
+            LOGGER.warning('Could not write attribute meta, because exposure'
+                           ' has only 1 data point')
+            self.meta = {}
```

Comment:
This part is a bit confusing. This is already handled in the `Exposure.__init__()` right?

Reply:
No this is not handled by init, in each exposure initiation method of an exposure class this is added manually.